core/src/main/java/org/stellar/anchor/sep38/Sep38Service.java

```diff
@@ -1,28 +1,119 @@
       Sep38Config sep38Config, AssetService assetService, RateIntegration rateIntegration) {
     this.sep38Config = sep38Config;
     this.assetService = assetService;
     this.rateIntegration = rateIntegration;
+    this.infoResponse = new InfoResponse(this.assetService.listAllAssets());
+    assetMap = new HashMap<>();
+    this.infoResponse.getAssets().forEach(asset -> assetMap.put(asset.getAsset(), asset));
     Log.info("Initializing sep38 service.");
   }
 
   public InfoResponse getInfo() {
-    List<AssetInfo> assets = this.assetService.listAllAssets();
-    return new InfoResponse(assets);
+    return this.infoResponse;
+  }
+
+  public GetPricesResponse getPrices(
+      String sellAssetName,
+      String sellAmount,
+      String countryCode,
+      String sellDeliveryMethod,
+      String buyDeliveryMethod)
+      throws AnchorException {
+    if (this.rateIntegration == null) {
+      throw new ServerErrorException("internal server error");
+    }
+    validateGetPricesInput(
+        sellAssetName, sellAmount, countryCode, sellDeliveryMethod, buyDeliveryMethod);
+
+    InfoResponse.Asset sellAsset = assetMap.get(sellAssetName);
+    assert sellAsset != null;
+
+    GetRateRequest.GetRateRequestBuilder builder =
+        GetRateRequest.builder()
+            .sellAsset(sellAssetName)
+            .sellAmount(sellAmount)
+            .countryCode(countryCode)
+            .sellDeliveryMethod(sellDeliveryMethod)
+            .buyDeliveryMethod(buyDeliveryMethod);
+    GetPricesResponse response = new GetPricesResponse();
+    for (String buyAssetName : sellAsset.getExchangeableAssetNames()) {
+      InfoResponse.Asset buyAsset = this.assetMap.get(buyAssetName);
+      if (!buyAsset.hasBuyDeliveryMethod(buyDeliveryMethod)) {
+        continue;
+      }
+
+      GetRateRequest request = builder.buyAsset(buyAssetName).build();
+      GetRateResponse rateResponse = this.rateIntegration.getRate(request);
+      response.addAsset(buyAssetName, rateResponse.getRate().getPrice());
+    }
+
+    return response;
+  }
+
+  public void validateGetPricesInput(
+      String sellAssetName,
+      String sellAmount,
+      String countryCode,
+      String sellDeliveryMethod,
+      String buyDeliveryMethod)
+      throws AnchorException {
+    Log.infoF(
+        "validateGetPricesInput(): sellAssetName={}, sellAmount={}, countryCode={}, sellDeliveryMethod={}, buyDeliveryMethod={}",
+        sellAssetName,
+        sellAmount,
+        countryCode,
+        sellDeliveryMethod,
+        buyDeliveryMethod);
+
+    if (Objects.toString(sellAssetName, "").isEmpty()) {
+      throw new BadRequestException("sell_asset cannot be empty");
+    }
+
+    InfoResponse.Asset sellAsset = assetMap.get(sellAssetName);
+    if (sellAsset == null) {
+      throw new NotFoundException("sell_asset not found");
+    }
+
+    if (Objects.toString(sellAmount, "").isEmpty()) {
+      throw new BadRequestException("sell_amount cannot be empty");
+    }
```

Comment:
💡 Could we add a check for whether or not `sellAmount` is a positive integer/decimal value?

Reply:
https://github.com/stellar/java-stellar-anchor-sdk/pull/113/commits/a91aa5988a8e921a8280427373bf7fb356e19ab3